src/languageservice/apis/schemaModification.ts

```diff
@@ -0,0 +1,92 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Red Hat, Inc. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+import { JSONSchemaService } from '../services/jsonSchemaService';
+import { JSONSchema } from '../jsonSchema04';
+
+export enum MODIFICATION_ACTIONS {
+    'delete',
+    'add'
+}
+
+export interface SchemaAdditions {
+    schema: string,
+    action: MODIFICATION_ACTIONS.add,
+    path: string,
+    key: string,
+    // tslint:disable-next-line: no-any
+    content: any
+}
+
+export interface SchemaDeletions {
+    schema: string,
+    action: MODIFICATION_ACTIONS.delete,
+    path: string,
+    key: string
+}
+
+export class SchemaModification {
```

Comment:
I am not sure why we have a new `api` folder for this file. `SchemaModification` is not really an API for the 3rd parties. If I am reading this correctly `YAMLSchemaService` together with `SchemaAdditions` and `SchemaDeletions` are the public 3rd party APIs for the functionality.